mirgecom/simutil.py

```diff
@@ -104,46 +104,94 @@ def exact_sim_checkpoint(discr, exact_soln, visualizer, eos, q,
     if do_viz is False and do_status is False:
         return 0
 
-    actx = q[0].array_context
-    nodes = thaw(actx, discr.nodes())
-    rank = 0
-
-    if comm is not None:
-        rank = comm.Get_rank()
+    if profiler is not None:
+        profiler.StartTimer(f"checkpoint_{step}")
```

Comment:
I'd be more comfortable leaving these out. For one, this seems to preempt https://github.com/illinois-ceesd/planning/issues/57, and for another, the non-PEP8 interface sticks out to me like a sore thumb.

Reply:
This is removed now, however...

I don't think it duplicates https://github.com/illinois-ceesd/planning/issues/57.   That Issue does not mention timing user-defined constructs - which I need.   Timing kernels only does not give the information that I need to profile the simulation.   For example, how do I use anything from https://github.com/illinois-ceesd/planning/issues/57 to time the I/O?  What if I want to time how long I spend inside the checkpoint routine (which I do)?

Reply:
(see the discussion there)